extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,162 @@
+      config.onConfiguration &&
+      typeof config.onConfiguration === 'function'
+    ) {
+      config = config.onConfiguration(config, {
+        params,
+        query,
+      });
+    }
+    this._addConfiguration(config);
+  }
+
+  public getAuthorizationHeader() {
+    const xhrRequestHeaders = {};
+    const authHeaders = this.userAuthenticationService.getAuthorizationHeader();
+    if (authHeaders && authHeaders.Authorization) {
+      xhrRequestHeaders.Authorization = authHeaders.Authorization;
+    }
+    return xhrRequestHeaders;
+  }
+
+  public generateWadoHeader(config) {
+    const authorizationHeader = this.getAuthorizationHeader();
+    //Generate accept header depending on config params
+    const formattedAcceptHeader = utils.generateAcceptHeader(
+      config.acceptHeader,
+      config.requestTransferSyntaxUID,
+      config.omitQuotationForMultipartRequest
+    );
+
+    return {
+      ...authorizationHeader,
+      Accept: formattedAcceptHeader,
+    };
+  }
+
+  // sets authorization headers before queries
+  public setQidoHeaders() {
+    this.clients.forEach(
+      client =>
+        (client.qidoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  public setWadoHeaders(tipo = 1) {
+    if (tipo === 1) {
+      this.clients.forEach(
+        client =>
+          (client.wadoDicomWebClient.headers = this.generateWadoHeader(client))
+      );
+    } else {
+      this.clients.forEach(
+        client =>
+          (client.wadoDicomWebClient.headers = this.getAuthorizationHeader())
+      );
+    }
+  }
+
+  // calls reject for each client that supports it
+  public reject() {
+    this.clients.forEach(client => {
+      if (client.supportsReject) {
+        dcm4cheeReject(client.wadoRoot);
+      }
+    });
+  }
+
+  // returns the qido client given the client name
+  public getQidoClient(name = undefined) {
+    if (this.clients.length) {
+      if (name) {
+        const client = this.clients.find(client => client.name === name);
+        return client.qidoDicomWebClient;
+      } else {
+        return this.clients[0].qidoDicomWebClient;
+      }
+    }
+  }
+
+  // returns the wado client given the client name
+  public getWadoClient(name = undefined) {
+    if (this.clients.length) {
+      if (name) {
+        const client = this.clients.find(client => client.name === name);
+        return client.wadoDicomWebClient;
+      } else {
+        return this.clients[0].wadoDicomWebClient;
+      }
+    }
+  }
+
+  // returns client configuration given the client name
+  public getClient(name = undefined) {
+    if (this.clients.length) {
+      if (name) {
+        const client = this.clients.find(client => client.name === name);
+        return client;
+      } else {
+        return this.clients[0];
+      }
+    }
+  }
+
+  // returns the client list
+  public getClients() {
+    return this.clients;
+  }
+
+  constructor({ params, query, dicomWebConfig, userAuthenticationService }) {
```

Comment:
constructors are typically at the top of the class... after private instance variables.